### PR TITLE
fix(forms): mark nested formGroups as submitted on submit (#21263)

### LINKED ISSUE #21263: FormGroupDirective.submitted is not propagated to inner FormGroupDirectives
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

#### What is the expected behavior?
Error should be shown even if I have more then one [formGroup] on a page.

#### What is the current behavior?
Error not shown.

#### What are the steps to reproduce?
1. https://stackblitz.com/edit/angular-evuu4c
2. Click submit
3. Address 2 should have an error message, but it doesn't.

Important piece is in input-form-example.html:36. I use `[formGroup]="form.get('address')"` instead of `formGroupName="address"`and that is causing the bug. Material component depend on FormGroupDirective.submitted property to show errors on fields after submit. That is set only on FormGroupDirective added to `form` tag itself, but not other FormGroupDirectives.

#### What is the use-case or motivation for changing an existing behavior?
In my case I have parts of my form in subcomponents, pretty complicated ones, and I cannot use formGroupName. As a result validation errors don't show.

In some places I started doing this:

```
// Tell internal form that outer form was submitted.
// Needed to show validation error even for untouched inputs.
this.outerForm.ngSubmit.pipe(takeUntil(this.destroy))
    .subscribe((event: Event) => {
        this.innerFormGroupDirective.onSubmit(event);
    });
```
but I think FormGroupDirective should just do that (if host is no the `form`).

### PR DESCRIPTION
**Pull request by ukrukarg**
Previously only formGroup on the <form> was marked as submitted, not also nested formGroup are marked.

Fixes #21263

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #21263

## What is the new behavior?
Previously only formGroup on the <form> was marked as submitted, not also nested formGroup are marked.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: packages/forms/src/directives/reactive_directives/form_group_directive.ts [hunk 1]
```diff
@@ -76,8 +77,12 @@ export class FormGroupDirective extends ControlContainer implements Form,
 
   constructor(
       @Optional() @Self() @Inject(NG_VALIDATORS) private _validators: any[],
-      @Optional() @Self() @Inject(NG_ASYNC_VALIDATORS) private _asyncValidators: any[]) {
+      @Optional() @Self() @Inject(NG_ASYNC_VALIDATORS) private _asyncValidators: any[],
```
Comment (Toxicable) [inline packages/forms/src/directives/reactive_directives/form_group_directive.ts]: Since `FormGroupDirective` this is a breaking change

Comment (ukrukarg) [reply packages/forms/src/directives/reactive_directives/form_group_directive.ts]: You're right, marked as breaking.

**Comment by ukrukarg** [comment]
Integration tests failed due to timeout, any way I can rerun them?

**Comment by kara** [comment]
@ukrukarg and I discussed offline that we want to pursue a different approach, so closing.

See https://github.com/angular/angular/issues/21263#issuecomment-357074997 for more info.

**Pull request closed without merging**